### PR TITLE
Make party ID mandatory & check validity

### PR DESCRIPTION
**Pull request by samaij**
* Move Party ID from address settings to basic settings
* Check Party ID validity when saving module settings form

### CODE REVIEW

**Comment by samaij** [comment]
For running just the validator tests:
```
./vendor/bin/phpunit --debug --verbose --filter testPartyIdIsValid tests/unit/ValidatorTest.php 
```

File: vg_postnord.php [hunk 1]
```diff
@@ -368,6 +371,13 @@ protected function getConfigForm(): array
                         'desc' => $this->trans('Get this information from Postnord. Something like abc123123123123abc123', [], 'Modules.Vgpostnord.Admin'),
                         'required' => true
                     ],
+                    [
+                        'type' => 'text',
+                        'name' => 'VG_POSTNORD_PARTY_ID',
+                        'label' => $this->trans('Party ID', [], 'Modules.Vgpostnord.Admin'),
+                        'desc' => $this->trans('Get this information from Postnord', [], 'Modules.Vgpostnord.Admin'),
```
Comment (tswfi) [inline vg_postnord.php:378]:
```suggestion
                        'desc' => $this->trans('10 digits long Party ID. Get this information from Postnord', [], 'Modules.Vgpostnord.Admin'),
```

File: vg_postnord.php [hunk 2]
```diff
@@ -813,6 +817,10 @@ protected function postProcess(): bool
             $result &= Configuration::updateValue($key, Tools::getValue($key));
         }
 
+        if (!VgPostnordPartyIdValidator::partyIdIsValid(Tools::getValue("VG_POSTNORD_PARTY_ID"))) {
+            $this->context->controller->errors[] = $this->trans("Party ID does not seem to be valid.", [], "Modules.Vgpostnord.Admin");
```
Comment (tswfi) [inline vg_postnord.php:821]:
```suggestion
            $this->context->controller->errors[] = $this->trans("Party ID is not valid.", [], "Modules.Vgpostnord.Admin");
```

File: vg_postnord.php (hunk 1 above)
Comment (samaij) [reply vg_postnord.php]: https://github.com/vilkasgroup/vg_postnord/pull/14/commits/4a193f32d32544740dcb7b530d11cf52ff220ad5

**Review by tswfi — APPROVED** [review]
👍 Great!

**Pull request merged**